### PR TITLE
Update Doorbell Siren 6 to support Sound/Volume control

### PR DESCRIPTION
**Pull request by Aeotec-ccheng**
- Added setting/preference option to configure volume/sound via Multichannel Cmd Encap (Sound Switch Configuration SET)

### CODE REVIEW

File: devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy [hunk 1]
```diff
@@ -66,10 +110,24 @@ def installed() {
...
+	if (SirenDoorbellSend == true) //keep Z-Wave traffic low, requires bool button in setting to trigger.
+	{
+    		soundControl(SirenDoorbellEndpoint, SirenDoorbellVolume, SirenDoorbellSound)
+	}
+}
+
+def soundControl(Endpoint, Volume, Sound) {
+	if (Endpoint && Volume && Sound) {
+    		log.debug "soundControl($Endpoint, $Volume, $Sound)"
+        	encap(zwave.multiChannelV3.multiChannelCmdEncap(sourceEndPoint:0, destinationEndPoint:Endpoint, commandClass:121, command:5, parameter: [Volume,Sound]))
```
Comment (cbaumler) [inline devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: I generally see command class values in hexadecimal.

Comment (Aeotec-ccheng) [reply devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: @cbaumler 

commandClass = 0x79 //Sound Switch Command Class
command = 0x05 //Configuration SET

Comment (greens) [reply devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: @Aeotec-ccheng he's just saying that our typical coding style is to write these params as hex in the code

Comment (greens) [reply devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: Also, isn't the intent of the `encap` method to handle this? I don't think you should be passing a multichannel encapsulated command to `encap`

Comment (Aeotec-ccheng) [reply devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: @greens - shouldn't be a problem to change these values to hex.

Comment (Aeotec-ccheng) [reply devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: i can change the method of how the multichannel cmd encap is handled, but it does need to be secure encapsulated.

I've modified private encap() to properly pass multichannel encapsulation, but if needed, I can move it to a different def/func.

**Review by greens — CHANGES_REQUESTED** [review]
A couple of general coding style fixes:

* zwave command classes should consistently be written as hex values
* device preferences and variables should not have initial letters capitalized
* open braces should go on the same line as if statements, and closing braces should be on the same line as following `else`s

**Comment by Aeotec-ccheng** [comment]
> A couple of general coding style fixes:
> 
> * zwave command classes should consistently be written as hex values
> * device preferences and variables should not have initial letters capitalized
> * open braces should go on the same line as if statements, and closing braces should be on the same line as following `else`s

Thanks, I'll work on this shortly to follow these coding styles.

File: devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy (hunk 1 above)
Comment (Aeotec-ccheng) [reply devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: @cbaumler @greens - Changed from decimal to hex value use and tested.

Separated Encap function to handle MC cmd encap(sound switch configuration set) separately:
private encap()
private mcEncap()

**Comment by Aeotec-ccheng** [comment]
@greens - Finished formatting based on code style, please let me know if any other changes are needed here.

File: devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy [hunk 2]
```diff
@@ -268,3 +314,29 @@ def keepChildrenOnline() {
...
+		} else {
+			cmd.format()
+		}
+	}
+}
+
+private mcEncap(cmd) {
+	if (cmd) {
+		if (zwaveInfo.zw.contains("s")) {
+			def rawZwaveData = zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
```
Comment (greens) [inline devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: okay, I think I know what you were running into. The context of the function is what was giving you trouble. Based on where you're calling this from, you should just be able to wrap this object in `response()` and return that

File: devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy [hunk 3]
```diff
@@ -98,15 +155,17 @@ def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap
 		cmd.command = cmd.parameter[1]
 		cmd.parameter = cmd.parameter.drop(2)
 	}
-	def encapsulatedCommand = cmd.encapsulatedCommand()
+    
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x60: 3, 0x79: 1])
```
Comment (greens) [inline devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: Can you remove the `0x79: 1` you're passing in?

We haven't updated our backend to support parsing of Sound Switch yet (and there's no handler established in this DTH anyway) so it's not actually doing anything.

Comment (Aeotec-ccheng) [reply devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: @greens - Got it, I've removed it from the encapsulatedCommand and secureEncapsulatedCommand.

File: devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy (hunk 2 above)
Comment (Aeotec-ccheng) [reply devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: I'll give this a quick test and change this in the next hour.

Comment (Aeotec-ccheng) [reply devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy]: @greens - Thanks that worked like a charm.

**Review by cbaumler — DISMISSED** [review]

**Comment by Aeotec-ccheng** [comment]
I noticed the other endpoints aren't triggering which is caused by state.lastTriggeredSound not changing when controlled via button.

Edited the Multichannel command class to change state.lastTriggeredSound only if NOTIFICATION CC is controlled.

**Comment by greens** [comment]
@MGoralczykS want to take a look here?

**Comment by MGoralczykS** [comment]
@greens This fixing issue ICP-12798. This looks good.

**Review by greens — APPROVED** [review]

**Pull request merged**